packages/world/src/systems/MineSystem.sol

```diff
@@ -93,93 +93,66 @@ contract MineSystem is System {
     caller.activate();
     caller.requireConnected(coord);
 
-    (EntityId mined, ObjectTypeId mineObjectTypeId) = getOrCreateEntityAt(coord);
-    require(mineObjectTypeId.isMineable(), "Object is not mineable");
+    (EntityId mined, ObjectTypeId minedType) = getOrCreateEntityAt(coord);
+    require(minedType.isMineable(), "Object is not mineable");
 
     mined = mined.baseEntityId();
     Vec3 baseCoord = Position._get(mined);
 
-    if (mineObjectTypeId.isMachine()) {
+    if (minedType.isMachine()) {
       (EnergyData memory machineData,) = updateMachineEnergy(mined);
       require(machineData.energy == 0, "Cannot mine a machine that has energy");
-    } else if (mineObjectTypeId.isSeed()) {
-      _requireSeedNotFullyGrown(mined);
-    } else if (mineObjectTypeId == ObjectTypes.AnyOre) {
-      mineObjectTypeId = RandomResourceLib._collapseRandomOre(mined, coord);
+    } else if (minedType == ObjectTypes.AnyOre) {
+      minedType = RandomResourceLib._collapseRandomOre(mined, coord);
     }
 
-    // First coord will be the base coord, the rest is relative schema coords
-    Vec3[] memory coords = mineObjectTypeId.getRelativeCoords(baseCoord, Orientation._get(mined));
+    MineLib._requireMinesAllowed(caller, minedType, coord, extraData);
```

Comment:
Are there any re-entrancy issues from calling this before the Mass table is updated?

Reply:
Actually `_collapseRandomOre` would get called multiple times right now because the type will remain any_ore, if mine is re-entered, but I think the type should be set and that would solve it

Reply:
nvm it does set it so it'd only be called once!

Reply:
I moved it to the end just to be sure